pandas/tests/extension/base/interface.py

```diff
@@ -33,6 +34,10 @@ def test_array_interface(self, data):
         result = np.array(data)
         assert result[0] == data[0]
 
+        result = np.array(data, dtype=object)
```

Comment:
I think worth generically testing this for all dtypes (as this is supposed to be a pass thru) is worth it

Reply:
See https://github.com/pandas-dev/pandas/pull/23593/files#r241486278

Reply:
And to repeat: We are testing `dtype` with `dtype=object` ? That's the only dtype for which we know what the result will be, for all other dtypes it depends on the array dtype.

Reply:
but we *do* know the result for other dtypes, its ``np.array(list(array), dtype=dtype)``

Reply:
otherwise if you change __array__ then you could easily regress

Reply:
These are the *base* tests, we cannot assume anything about the expected result here. Each array type can still test the specifics for different dtypes in its own tests, which is what I am doing in this PR for Datetime/Timedelta/PeriodArray.

Reply:
> but we do know the result for other dtypes, its np.array(list(array), dtype=dtype)

And to come back to this specific example: that is not generally True. Eg for DatetimeArray, `list(array)` will give a list of Timestamp objects, which numpy cannot convert to any other dtype:

```
In [49]: a = pd.arrays.DatetimeArray(pd.array([1, 2], dtype='datetime64[ns]')._ndarray)

In [50]: a                                                                                         
Out[50]: 
<DatetimeArray>
['1970-01-01 00:00:00.000000001', '1970-01-01 00:00:00.000000002']
Length: 2, dtype: datetime64[ns]

In [51]: list(a)                                                                             
Out[51]: 
[Timestamp('1970-01-01 00:00:00.000000001'),
 Timestamp('1970-01-01 00:00:00.000000002')]

In [53]: np.array(list(a), dtype='float')                                                        
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-53-1a9c1baaa853> in <module>
----> 1 np.array(list(a), dtype='float')

TypeError: float() argument must be a string or a number, not 'Timestamp'
```

Reply:
@jreback OK?

Reply:
its fine i guess.